infra/gcp/dns/dns.tf

```diff
@@ -1,7 +1,7 @@
 // knative.dev DNS
 module "knative_dev" {
   source     = "terraform-google-modules/cloud-dns/google"
-  version    = "3.0.0"
+  version    = "~> 4.1"
```

Comment:
DO we want an autobumper for this at some point?

Reply:
Not for terraform modules. Major changes almost always require some manual changes.